README.md

```diff
@@ -1,2 +1,70 @@
-# scraped_page
+# ScrapedPage
+
 Write declarative scrapers in Ruby
+
+## Usage
+
+Create a subclass of `ScrapedPage` for each page you wish to scrape.
+
+```ruby
+require 'scraped_page'
+
+class ExamplePage < ScrapedPage
+  field :title do
+    noko.at_css('h1').text
+  end
+
+  field :more_information do
+    noko.at_css('a')[:href]
+  end
+end
+```
+
+Then you can create a new instance and pass in a url
+
+```ruby
+page = ExamplePage.new(url: 'http://example.com')
+
+page.title
+# => "Example Domain"
+
+page.more_information
+# => "http://www.iana.org/domains/reserved"
+
+page.to_h
+# => { :title => "Example Domain", :more_information => "http://www.iana.org/domains/reserved" }
+```
+
+## Archiving scraped pages
+
+The default strategy for retrieving pages is `ScrapedPage::Strategy::LiveRequest`. If you'd also like to archive a copy of the page you're scraping into a git branch, you can pass a `:strategy` option to the constructor:
+
+```ruby
+ExamplePage.new(url: 'http://example.com', strategy: ScrapedPage::Strategy::LiveRequestArchive.new)
+```
+
+This will use the [`scraped_page_archive`](https://github.com/everypolitician/scraped_page_archive)
+gem to store a copy of the pages you scrape in a git branch in your scraper's repo.
+
+## Custom strategies
+
+If for some reason you can't scrape a site using one of the build in strategies
+then you can provide your own strategy.
+
+A strategy is an object that responds to a `get` method and returns a
+`ScrapedPage::Response`.
+
+```ruby
+class FilesystemStrategy
+  def get(url)
```

Comment:
I don't think `get` is the best name for this method. At the moment we're (purposely) not providing any way for people to make a `POST` request, they'd need to make their own custom strategy. But having a strategy that makes a `POST` request in a method called `get` seems a bit wrong.

I toyed with using `request` as the method name, but then in lots of Ruby libraries that has the method signature `request(method = :get, url)`.

Perhaps something like `fetch`?